nvtabular/io.py

```diff
@@ -405,6 +406,12 @@ def set_col_names(self, labels, cats, conts):
         self.labels = labels
         self.column_names = labels + conts
 
+    def set_col_idx(self, gdf):
```

Comment:
Why do we need to use the gdf for this? Can we use the cats/conts/labels columns to define this mapping instead? I'm asking because we now need 3 function calls to initialize the hugectr writer properly (constructor, set_column_names, set_col_idx), and the API is pretty unwieldy

Reply:
We want to know in which order are the columns in the gdf, so cats/conts/labels list cannot provide that information.

I agree with the API. I can gdf as a parameter of set_col_names and do all in that function. What do you think?

Reply:
How about just lazily initializing this? Like initialize on the first call to add_data if its not initialized already?

Reply:
Sounds good to me. @oyilmaz-nvidia for confirming.

Reply:
Yeah, we can do that as well.